### ISSUE TITLE
`padding_mode` support for `functional.unfold`

### ISSUE THREAD

**Issue by siarez** [pytorch/pytorch#27037]
## 🚀 Feature
Add `padding_mode` support for `functional.unfold` operation.

## Motivation
`unfold` is a very useful function for window operations other than convolution. In its current implementation, it only supports the "constant" padding mode, and automatically fills the margins with zeros. I have a use case where I need to use the "edge" padding mode. 

## Pitch

I want to `unfold` to have the `padding_mode` argument exactly like the convolution operation. This would also make the API more coherent.

## Alternatives

I guess the alternative is to manually copy cell values to the padded area.

**Comment by siarez** [comment]
I just realized `nn.functional.pad` function exists and can be used to solve this problem, without building in `padding_mode` into `unfold`. In light of this, maybe this issue can be closed.
